Bottom: cc33ae3210314bebef8714f90ba31f1c384cc151
Top:    02204c0b5d6e6b510ee03db52cca7823ef4bab5a
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-05 13:58:16 +0800

autofs-5.0.7 - fix spawn_umount() return check in mount_bind.c:lookup_init()

Check return of spawn_umount() and report it if it fails.


---

diff --git a/modules/mount_bind.c b/modules/mount_bind.c
index 4975294..d6c6fe7 100644
--- a/modules/mount_bind.c
+++ b/modules/mount_bind.c
@@ -57,7 +57,8 @@ int mount_init(void **context)
 		bind_works = 1;
 	}
 
-	spawn_umount(LOGOPT_NONE, "-n", t2_dir, NULL);
+	if (spawn_umount(LOGOPT_NONE, "-n", t2_dir, NULL) != 0)
+		debug(LOGOPT_ANY, MODPREFIX "umount failed for %s", t2_dir);
 
 out:
 	rmdir(t1_dir);
